build/tasks/AddArchiveReferences.cs

```diff
@@ -52,12 +52,27 @@ public override bool Execute()
             foreach (var package in archiveArtifacts)
             {
                 var packageName = package.ItemSpec;
-                var packageVersion = string.Equals(packageName, "Microsoft.AspNetCore.All", StringComparison.OrdinalIgnoreCase) ?
-                    MetaPackageVersion :
-                    buildArtifacts
-                        .Single(p => string.Equals(p.PackageInfo.Id, packageName, StringComparison.OrdinalIgnoreCase))
-                        .PackageInfo.Version.ToString();
 
+                string packageVersion;
+
+                if(string.Equals(packageName, "Microsoft.AspNetCore.All", StringComparison.OrdinalIgnoreCase))
```

Comment:
@JunTaoLuo remind me again why this is special cased? And does this need to be updated to take into consideration that we have 2 metapackages?

Reply:
Yup, I think it's gonna need to include .App as well. The reason this is here is that the metapackages are not included in the build artifacts collected from modules so it wasn't possible to determine what version should be referenced when building the LZMA. To resolve this, we pass in the metapackage version directly from the msbuild invocation.

Reply:
Makes since. This is probably why the PackageArchive build step is currently failing.

Reply:
My bad, totally forgot about this.